sound/soc/sof/topology.c

```diff
@@ -2250,18 +2256,6 @@ static int sof_link_hda_load(struct snd_soc_component *scomp, int index,
 		}
 	}
 
-	/* set config for all DAI's with name matching the link name */
-	ret = sof_set_dai_config(sdev, size, link, config);
-	if (ret < 0) {
-		if (link->dpcm_playback)
-			dev_err(sdev->dev, "error: failed to save DAI config for playback HDA%d\n",
-				config->dai_index);
-
-		if (link->dpcm_capture)
-			dev_err(sdev->dev, "error: failed to save DAI config for capture HDA%d\n",
-				config->dai_index);
-	}
-
```

Comment:
I am lost as to why this is removed? doesn't that function do precisely what you wanted in the first place?

Reply:
As state before, the HDA dai need match both name and direction, sof_set_dai_config is a common API which only check the name. So the dai config setting part is move into the sof_link_hda_process, where the name and direction both checked.

Reply:
please look into extending it to cover direction then. No need for duplication of code.

Reply:
@zhuyingjiang you could do some other checks in the function to check the type as well....if type if hda, check name and direction, else just name is enough. make sense?

Reply:
@RanderWang what is your opinion? since you write the sof_set_dai_config, how to refine this is the better way?

Reply:
@RanderWang OK, please go ahead refine your code, please also consider the two dai_index Pierre mentioned.

Reply:
ok, I will consider my plan again. I don't want to make a aggressive change without great improvement